src/plugins/index_pattern_management/public/service/index_pattern_management_service.ts

```diff
@@ -37,31 +31,26 @@ interface SetupDependencies {
  * @internal
  */
 export class IndexPatternManagementService {
-  public setup({ httpClient, home }: SetupDependencies) {
-    const creation = new IndexPatternCreationManager(httpClient);
-    const list = new IndexPatternListManager();
+  indexPatternCreationManager?: IndexPatternCreationManager;
+  indexPatternListConfig?: IndexPatternListManager;
 
-    creation.add(IndexPatternCreationConfig);
-    list.add(IndexPatternListConfig);
+  public setup({ httpClient }: SetupDependencies) {
+    this.indexPatternCreationManager = new IndexPatternCreationManager(httpClient);
+    this.indexPatternListConfig = new IndexPatternListManager();
 
-    home.featureCatalogue.register({
-      id: 'index_patterns',
-      title: i18n.translate('management.indexPatternHeader', {
-        defaultMessage: 'Index Patterns',
-      }),
-      description: i18n.translate('management.indexPatternLabel', {
-        defaultMessage:
-          'Manage the index patterns that help retrieve your data from Elasticsearch.',
-      }),
-      icon: 'indexPatternApp',
-      path: '/app/kibana#/management/kibana/index_patterns',
-      showOnHomePage: true,
-      category: FeatureCatalogueCategory.ADMIN,
-    });
+    this.indexPatternCreationManager.setup.addCreationConfig(IndexPatternCreationConfig);
+    this.indexPatternListConfig.setup.addListConfig(IndexPatternListConfig);
 
     return {
-      creation,
-      list,
+      creation: this.indexPatternCreationManager.setup,
```

Comment:
nit: it's a little bit unusual because in other places we execute the `setup` method instead of passing reference like you did. first of all it allows us to: 
1) don't bind functions to service context. See above e.g.:
`getType: this.getType.bind(this),`
2) we can create instance of service in constructor and pass `setup/start` dependencies directly. In your example you create instance in setup hook as a result in start you should use : `'!'`
`creation: this.indexPatternCreationManager!.start,`